content/tutorials/publish-subscribe.textile

```diff
@@ -14,339 +14,7 @@ Messages published can contain string, JSON object, JSON array or binary data pa
-
-  "See this step in Github":https://github.com/ably/tutorials/commit/277475e
-
-blang[swift].
-  We will start by creating an Xcode project for this tutorial. To build your own Xcode Project in Swift visit "Apple developer website":https://developer.apple.com/library/content/documentation/IDEs/Conceptual/AppStoreDistributionTutorial/Setup/Setup.html and get familiar with steps necessary to setup your own application.
-  When you setup your application delete the default @ViewController.swift@ add new File -> New -> File... and choose @Cocoa Touch Class@.
-  <a href="/images/tutorials/tutorials-swift-CTC.png" target="_blank">
-    <img src="/images/tutorials/tutorials-swift-CTC.png" style="width: 100%" alt="Create new Cocoa Touch Class">
-  </a>
-  Name your new class "ExampleViewController" and choose @Swift@ as language:
-  <a href="/images/tutorials/tutorials-swift-ExampleClass.png" target="_blank">
-    <img src="/images/tutorials/tutorials-swift-ExampleClass.png" style="width: 100%" alt="Name new Cocoa Touch Class">
-  </a>
-  After that navigate to @Main.storyboard@ in your project, click on the @ViewController@ that has already been added by default during project creation and from the @Utilities@ that are located on the right choose @Identity Inspector@. Find the field labeled "Class" and select "ExampleViewController".
-  <a href="/images/tutorials/tutorials-swift-IB-class.png" target="_blank">
-    <img src="/images/tutorials/tutorials-swift-IB-class.png" style="width: 100%" alt="Interface design">
-  </a>
-
-  "See this step in Github":https://github.com/ably/tutorials/commit/a98f1bf
-
-  To start using Ably you first need to install the Ably pod via CocoaPods. You need to add a @Podfile@ to your project directory:
-
-  bc[sh]. touch Podfile
-
-  Then add this line to your application’s Podfile:
-
-  ```[text]
-    pod 'Ably'
-  ```
-
-  Install it with:
-
-  bc[sh]. pod install
-
-  To learn more about using CocoaPods in your project visit "official CocoaPods guide":https://guides.cocoapods.org/using/using-cocoapods.html.
-
-  Then in your files in which you will be using Ably import:
-
-  ```[swift]
-    import Ably
-  ```
-
-  To connect to Ably, you need to instance the client library with the API key you copied in Step 1. API keys used with "basic authentication":/general/authentication#basic-authentication for your own servers is generally preferred, however clients running on insecure devices should always use the "token authentication scheme":/general/authentication#token-authentication instead. In this example, we use an API key for simplicity.
-
-  Add the following to the file in which you imported the Ably library:
-
-  ```[swift]
-    let API_KEY = "INSERT-YOUR-API-KEY-HERE"
-    let client = ARTRealtime(key: API_KEY)
-  ```
-
-  "See this step in Github":https://github.com/ably/tutorials/commit/59dcc56
+<%= partial 'tutorials/_install_ablyrealtime', vars(why_realtime: "subscribe to messages") %>
```

Comment:
I'm happy to expand this back out into duplicates, but TBH I'm a bit surprised to see you object to this, seemed like an obvious DRYing up. Having other languages rendered, is fixable with a couple lines of `<% if js %> ... <% end %>` etc. This is a pretty basic partial and we're only going to add more tutorials, do we really want to end up with more or less the same 330 lines of code copied and pasted into a dozen different ones?